kernel-switcher/test_in_lxd_vm.py

```diff
@@ -0,0 +1,155 @@
+#!usr/bin/env python3
+"""
+This is a test script for the kernel switcher.
+It uses LXD to create a VM in which the test script runs this sequence:
+- install a kernel that's older than the default
+- reboot
+- check that the old kernel is _still not_ running
+- use the kernel switch tool to switch to the old kernel
+- reboot
+- check that the older kernel is running
+"""
+import subprocess
+import shlex
+import time
+import contextlib
+import signal
+
+
+class InterruptHandler:
+    def __init__(self):
+        self.interrupted = False
+        self.first_interrupt_time = time.time()
+        signal.signal(signal.SIGINT, self.signal_handler)
+
+    def signal_handler(self, signum, frame):
+        if self.interrupted and time.time() > self.first_interrupt_time + 2:
+            print("\nForced exit.You'll have to do the cleanup yourself.")
+            raise KeyboardInterrupt
+        self.interrupted = True
+        print(
+            "\nCTRL+C detected while using a VM...\n"
+            "Press CTRL+C again in 2 seconds if you really want to force exit"
+        )
+
+
+interrupt_handler = InterruptHandler()
```

Comment:
Why do we have this mechanism?

Reply:
To handle cleanup gracefully, double tapping ctrl+c would kill the vm disposal.